src/Microsoft.Extensions.HealthChecks/HealthCheckService.cs

```diff
@@ -0,0 +1,86 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.HealthChecks.Internal;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+
+namespace Microsoft.Extensions.HealthChecks
+{
+    /// <summary>
+    /// Default implementation of <see cref="IHealthCheckService"/>
+    /// </summary>
+    public class HealthCheckService : IHealthCheckService
+    {
+        private readonly ILogger<HealthCheckService> _logger;
+
+        /// <summary>
+        /// A <see cref="IReadOnlyDictionary{TKey, T}"/> containing all the health checks registered in the application.
+        /// </summary>
+        /// <remarks>
+        /// The key maps to the <see cref="IHealthCheck.Name"/> property of the health check, and the value is the <see cref="IHealthCheck"/>
+        /// instance itself.
+        /// </remarks>
+        public IReadOnlyDictionary<string, IHealthCheck> Checks { get; }
+
+        /// <summary>
+        /// Constructs a <see cref="HealthCheckService"/> from the provided collection of <see cref="IHealthCheck"/> instances.
+        /// </summary>
+        /// <param name="healthChecks">The <see cref="IHealthCheck"/> instances that have been registered in the application.</param>
+        public HealthCheckService(IEnumerable<IHealthCheck> healthChecks) : this(healthChecks, NullLogger<HealthCheckService>.Instance) { }
+
+        /// <summary>
+        /// Constructs a <see cref="HealthCheckService"/> from the provided collection of <see cref="IHealthCheck"/> instances, and the provided logger.
+        /// </summary>
+        /// <param name="healthChecks">The <see cref="IHealthCheck"/> instances that have been registered in the application.</param>
+        /// <param name="logger">A <see cref="ILogger{T}"/> that can be used to log events that occur during health check operations.</param>
+        public HealthCheckService(IEnumerable<IHealthCheck> healthChecks, ILogger<HealthCheckService> logger)
+        {
+            Checks = healthChecks.ToDictionary(c => c.Name);
```

Comment:
This will throw on duplicate keys, and I'm not sure it's easy to diagnose when that happens.

Reply:
Yep, had the same thought. Not sure the best way to handle that. We can preemptively detect it and throw a clearer exception, but other than that I'm not sure if we can do much better given our DI-based approach.

Reply:
Yeah, I'd suggest when you build the real thing to just throw a better message. I don't believe the default duplicatekeyexception even prints the key